netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+                packageId = item.GetMetadata("ParentPackage");
+
+                var versionSeperatorIndex = packageId.IndexOf('/');
+
+                if (versionSeperatorIndex != -1)
+                {
+                    packageId = packageId.Substring(0, versionSeperatorIndex);
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(packageId) && PreferredPackages != null)
+            {
+                EnsurePackageRanks();
+
+                packageRanks.TryGetValue(packageId, out rank);
+            }
+
+            return rank;
+        }
+
+        private static string GetSourcePath(ITaskItem item)
+        {
+            var sourcePath = item.GetMetadata("HintPath");
+
+            if (String.IsNullOrWhiteSpace(sourcePath))
+            {
+                sourcePath = item.ItemSpec;
+            }
+
+            return sourcePath;
+        }
+
+        static readonly string[] s_targetPathMetadata = new[] { "TargetPath", "Path" };
+        private static string GetTargetPath(ITaskItem item)
+        {
+            // first use TargetPath, then Path, then fallback to filename+extension alone
+            foreach (var metadata in s_targetPathMetadata)
+            {
+                var value = item.GetMetadata(metadata);
+
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    return value.Replace('\\', '/');
+                }
+            }
+
+            var sourcePath = GetSourcePath(item);
+
+            return Path.GetFileName(sourcePath);
+        }
+
+        private ITaskItem HandleConflict(ITaskItem item1, ITaskItem item2)
+        {
+            var conflictMessage = $"Encountered conflict between {item1.ItemSpec} and {item2.ItemSpec}.";
+
+            var sourcePath1 = GetSourcePath(item1);
+            var sourcePath2 = GetSourcePath(item2);
+
+            var exists1 = File.Exists(sourcePath1);
+            var exists2 = File.Exists(sourcePath2);
+
+            if (!exists1 || !exists2)
+            {
+                var fileMessage = !exists1 ?
+                                    !exists2 ? 
+                                      "both files do" :
+                                      $"{item1.ItemSpec} does" :
+                                  $"{item2.ItemSpec} does";
+
+                Log.LogMessage($"{conflictMessage}.  Could not determine winner because {fileMessage} not exist.");
+                return null;
+            }
+            
+            var assemblyVersion1 = TryGetAssemblyVersion(sourcePath1);
+            var assemblyVersion2 = TryGetAssemblyVersion(sourcePath2);
+
+            // if only one is missing version
+            if (assemblyVersion1 == null ^ assemblyVersion2 == null)
+            {
+                var nonAssembly = assemblyVersion1 == null ? item1.ItemSpec : item2.ItemSpec;
+                Log.LogMessage($"{conflictMessage}. Could not determine a winner because {nonAssembly} is not an assembly.");
```

Comment:
should these be warnings? 

one rule of thumb: the user must laways have a way to fix any warning and ideally the message suggests how. if they can't - can't be warning.